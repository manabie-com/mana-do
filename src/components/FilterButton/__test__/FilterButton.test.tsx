import React from 'react'
import { render } from "@testing-library/react";

import { FilterButton } from '../index'

import "@testing-library/jest-dom";

/**
 * Test cases
 * 1. should be rendered without crashing
 * 2. should be contain active class if active = true
 */

describe('<FilterButton />', () => {

  test('Should be rendered without crashing', () => {
    const wrapper = render(<FilterButton>test</FilterButton>)
    expect(wrapper.container).toMatchSnapshot()
  })

  test('should be contain active class if active = true', () => {
    const wrapper = render(<FilterButton active>test</FilterButton>)
    expect(wrapper.getByRole('button')).toHaveClass('active')
  })

})