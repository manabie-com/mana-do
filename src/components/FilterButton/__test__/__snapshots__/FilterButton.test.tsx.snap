// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterButton /> Should be rendered without crashing 1`] = `
<div>
  <button
    class="Action__btn"
    data-testid="filter-button"
  >
    test
  </button>
</div>
`;
